sdk/core/azure-core/inc/azure/core/tracing/tracing.hpp

```diff
@@ -3,309 +3,47 @@
 
 /**
  * @file
- * @brief Handling log messages from Azure SDK.
+ * @brief Public TracerProvider type used to represent a tracer provider.
  */
 
 #pragma once
 
-#include "azure/core/datetime.hpp"
-#include "azure/core/internal/extendable_enumeration.hpp"
-#include "azure/core/nullable.hpp"
-#include "azure/core/url.hpp"
-#include <array>
 #include <memory>
 #include <string>
-#include <vector>
-
-// Forward declare Azure::Core::Http::Request to resolve an include file ordering problem.
-namespace Azure { namespace Core { namespace Http {
-  class Request;
-}}} // namespace Azure::Core::Http
 
 namespace Azure { namespace Core { namespace Tracing {
-
+  class TracerProvider;
   namespace _internal {
-
-    /** The set of attributes to be applied to a Span or Event.
-     *
-     * @details
-     * An AttributeSet represents a set of attributes to be added to a span or
-     * event.
-     *
-     * @note Note that AttributeSets do *NOT* take a copy of their input values,
-     * it is the responsibility of the caller to ensure that the object remains
-     * valid between when it is added to the AttributeSet and when it is added to
-     * a span or event.
-     *
-     * OpenTelemetry property bags can hold:
-     *    - bool
-     *    - int32_t
-     *    - int64_t
-     *    - uint64_t
-     *    - double
-     *    - const char *
-     *    - std::string/std::string_view                      *** Not Implemented
-     *    - std::span<const bool>                             *** Not Implemented
-     *    - std::span<const int32_t>                          *** Not Implemented
-     *    - std::span<const int64_t>                          *** Not Implemented
-     *    - std::span<const uint32_t>                         *** Not Implemented
-     *    - std::span<const double>                           *** Not Implemented
-     *    - std::span<std::string/std::string_view>           *** Not Implemented
-     *    - uint64_t (not fully supported)                    *** Not Implemented
-     *    - std::span<uint64_t> (not fully supported)         *** Not Implemented
-     *    - std::span<const uint8_t> (not fully supported)    *** Not Implemented.
-     *
-     */
-    class AttributeSet {
-    public:
-      /**
-       * @brief Adds a Boolean attribute to the attribute set.
-       *
-       * @param attributeName Name of attribute to add.
-       * @param value Value of attribute.
-       */
-      virtual void AddAttribute(std::string const& attributeName, bool value) = 0;
```

Comment:
I am going to assume all this is copy-pasted verbatim into `tracing_impl.hpp`, or at least isn't materially different, to warrant another review.

Reply:
Yes.